### PR TITLE
Add support in Commons CSV for tracking byte positions during parsing.

### PR DESCRIPTION
**Pull request by DarrenJAN**
Summary of Modifications

1. **Test Data Files**: Added new test data files, and updated pom.xml to exclude these files from RAT checks, avoiding unapproved license checks.
2. **CSVParser class**:
 Constructor Enhancements
a. Added support for an optional parameter -- **String encoding**--, which specifies the encoding to use for the reader.
4. **CSVRecord class**
- private long characterByte:  start byte position of this record 
Add new Constructor: support track byte positions in record class 
5. **ExtendedBufferedReader Class**:

- private long bytesRead: Tracks the number of bytes read so far.
- private long bytesReadMark: Stores the marked byte position.
- CharsetEncoder encoder: Encoder used to calculate byte size of characters.
- getCharBytes(int current): This function calculates character bytes based on UTF-8 encoding. Note: it only supports UTF-8 due to the encoding algorithm used. Full encoding can be supported and we just need more effort on this. 
- reset() and mark() Methods: Enhanced to prevent consuming characters and bytes unintentionally.

Test result:
mvn 
image
image
Pass unit tests and other restrictions

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -28,6 +34,7 @@
   <url>https://commons.apache.org/proper/commons-csv/</url>
```
Comment (vzhang-progress) [inline pom.xml:34]: The version here on line 32 should be our corresponding version like 1.12.1-marklogic

File: pom.xml [hunk 2]
```diff
@@ -15,6 +15,12 @@
  See the License for the specific language governing permissions and
  limitations under the License.
 -->
+
+<!--
+
+ Modifications copyright © 2022, 2024 MarkLogic Corporation.
```
Comment (vzhang-progress) [inline pom.xml]: I believe Matt modified this file as well in 2017

**Review by yunzvanessa — CHANGES_REQUESTED** [review]

**Review by yunzvanessa — APPROVED** [review]

**Comment by DarrenJAN** [comment]
1. Needs to replace the jar and regression test before merge
2. Remember to squash and merge

**Comment by DarrenJAN** [comment]
There is a functional bug in the mlcp that needs to be addressed

**Pull request merged**